From fc7f79288ba28e82d04153763023b69755d6c3a1 Mon Sep 17 00:00:00 2001
From: Abhinav Kumar <abhikuma@codeaurora.org>
Date: Wed, 28 Feb 2018 17:21:38 +0530
Subject: qcacld-2.0: Fix check for num_ac in iface stats event handler

In function wma_unified_link_iface_stats_event_handler, num_ac
is received from the firmware and is used in the loop to populate
values into result. However the memory for result  is allocated
for WIFI_AC_MAX (which is 4). While copying the data from fw, the
fw loop is looped from i = 0 to i < num_ac which is 0 to 3.
So the value of num_ac = 4 from the fw is not an issue.

Modify the upper limit check for num_ac to be greater than
WIFI_AC_MAX instead of greater than equal to WIFI_AC_MAX.

Change-Id: Iefb71563e4608e6fe3ab300bf7a9a96f748c7853
CRs-Fixed: 2197481
---
 CORE/SERVICES/WMA/wma.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CORE/SERVICES/WMA/wma.c b/CORE/SERVICES/WMA/wma.c
index 335b8e5..2668939 100644
--- a/CORE/SERVICES/WMA/wma.c
+++ b/CORE/SERVICES/WMA/wma.c
@@ -4981,7 +4981,7 @@ static int wma_unified_link_iface_stats_event_handler(void *handle,
 		WMA_LOGA("%s: Invalid param_tlvs for Iface Stats", __func__);
 		return -EINVAL;
 	}
-	if (link_stats->num_ac >= WIFI_AC_MAX) {
+	if (link_stats->num_ac > WIFI_AC_MAX) {
 		WMA_LOGE("%s: Excess data received from firmware num_ac %d",
 			__func__, link_stats->num_ac);
 		return -EINVAL;
-- 
cgit v1.1

